src/app/ClusterStateCache.h

```diff
@@ -363,6 +530,10 @@ class AttributeCache : protected ReadClient::Callback
     NodeState mCache;
     std::set<ConcreteAttributePath> mChangedAttributeSet;
     std::vector<EndpointId> mAddedEndpoints;
+
+    std::set<EventData, EventDataCompare> mEventDataCache;
+    EventNumber mHighestReceivedEventNumber = 0;
```

Comment:
Still feels like we should have a way to init the cache with a given highest event number, if we know we have already seen it... Followup is OK for this part.

Reply:
Ah missed that. Will add.

Reply:
Added + tests.